### DISCUSSION TITLE
Limit Models Available to Attach in Relation Manager

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#3588] [General]
Is it possible to limit what models are available (and shown in the dropdown) to attach via a relation manager?

**Comment by danharrin** [comment]
You want to apply a scope to the query or something?

**Reply by intrepidws** [reply]
Exactly, yes.

**Reply by danharrin** [reply]
#3590

`AttachAction::make()->recordSelectOptionsQuery(fn ($query) => $query->where(...))`

**Reply by intrepidws** [reply]
Thanks @danharrin , appreciated as always.

**Reply by intrepidws** [reply]
Seriously, this was a huge help and I really appreciate you adding this functionality.

**Reply by danharrin** [reply]
You're welcome :)

**Reply by leekaimun** [reply]
This is great help. Another question - I want to limit the selection option based on the parent record id. How do I get the current parent's model id in the closure?

**Reply by leekaimun** [reply]
Just to answer my own question, I could use $livewire parameter to get the ownerRecord. Eg `$livewire->ownerRecord->id`.
